sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -281,6 +281,10 @@ private void populateDisplayData(DisplayData.Builder builder) {
       HashSet<PipelineOptionSpec> specs = new HashSet<>(optionsMap.get(option.getKey()));
 
       for (PipelineOptionSpec optionSpec : specs) {
+        if (!optionSpec.shouldSerialize()) {
```

Comment:
Can you add a comment here and below on what this check on `shouldSerialize` is doing? It is unclear why we would want to skip options.